pilot/pkg/config/kube/crd/controller/controller.go

```diff
@@ -292,13 +302,48 @@ func (c *controller) HasSynced() bool {
 
 func (c *controller) Run(stop <-chan struct{}) {
 	log.Infoa("Starting Pilot K8S CRD controller")
+
+	wg := sync.WaitGroup{}
+	wg.Add(len(c.kinds))
+
 	go func() {
+		// We wait for each kind to either start or determine the CRD does not exist
+		wg.Wait()
 		cache.WaitForCacheSync(stop, c.HasSynced)
 		c.queue.Run(stop)
 	}()
 
 	for _, ctl := range c.kinds {
-		go ctl.informer.Run(stop)
+		ctl := ctl
+		go func() {
+			if found, _ := c.client.CRDExists(ctl.schema); found {
+				ctl.runningMu.Lock()
+				ctl.running = true
+				ctl.runningMu.Unlock()
+				wg.Done()
+				ctl.informer.Run(stop)
+			} else {
+				log.Warnf("CRD %v not found; waiting for it to be created", ctl.schema.Name())
+				wg.Done()
+				delay := time.Second
+				maxDelay := time.Minute
+				for {
+					if found, _ := c.client.CRDExists(ctl.schema); found {
+						break
+					}
+					time.Sleep(delay)
+					delay *= 2
+					if delay > maxDelay {
+						delay = maxDelay
+					}
+				}
+				log.Infof("CRD %v found; starting informer", ctl.schema.Name())
```

Comment:
If timeout before the crd created, then it will ignore it silently? This is not a good ux

Reply:
it is never timing out

Reply:
IC， Add a log to indicate what is it doing